src/main/java/Duke.java

```diff
@@ -1,10 +1,226 @@
+            if (isEmptyDescription(todoInfo)) {
+                throw new InvalidCommandException();
+            }
+            tasks.add(new Todo(todoInfo));
+        } else if  (trimmedCommand.startsWith("deadline")) {
+            String[] deadlineInfo = trimmedCommand.replace("deadline", "").split("/by");
+            if (isEmptyDescription(deadlineInfo[0].trim()) || deadlineInfo.length != 2) {
+                throw new InvalidCommandException();
+            }
+            String description = deadlineInfo[0].trim();
+            String datetime = deadlineInfo[1].trim();
+            tasks.add(new Deadline(description, datetime));
+        } else if (trimmedCommand.startsWith("event")) {
+            String[] eventInfo = trimmedCommand.replace("event", "").split("/at");
+            if (isEmptyDescription(eventInfo[0].trim()) || eventInfo.length != 2) {
+                throw new InvalidCommandException();
+            }
+            String description = eventInfo[0].trim();
+            String datetime = eventInfo[1].trim();
+            tasks.add(new Event(description, datetime));
+        } else {
+            throw new UnknownCommandException();
+        }
+
+        printHorizontalLine();
+        System.out.println("\t Got it. I've added: this task:");
+        System.out.println("\t   " + tasks.get(tasks.size() - 1));
+        System.out.println("\t Now you have " + tasks.size() + " task(s) in the list.");
+        printHorizontalLine();
+        updateFile();
+    }
+
+    public static void deleteTask(int taskNum) {
+        if (taskNum <= 0 || taskNum > tasks.size()) {
+            System.out.println("\tInvalid task number!");
+            return;
+        }
+
+        printHorizontalLine();
+        System.out.println("\t Noted. I've removed this task:");
+        System.out.println("\t   " + tasks.remove(taskNum - 1));
+        System.out.println("\t Now you have " + tasks.size() + " task(s) in the list.");
+        printHorizontalLine();
+        updateFile();
+    }
+
+    public static boolean isEmptyDescription(String description) {
+        return ("".equals(description));
+    }
+
+    public static void markAsDone(int taskNum) {
+        if (taskNum <= 0 || taskNum > tasks.size()) {
+            System.out.println("\tInvalid task number!");
+            return;
+        } else if (tasks.get(taskNum - 1).isDone()) {
+            System.out.println("\tTask already done!");
+            return;
+        }
+        tasks.get(taskNum - 1).setDone(true);
+        printHorizontalLine();
+        System.out.println("\t Nice! I've marked this task as done:");
+        System.out.println("\t   " + tasks.get(taskNum - 1));
+        printHorizontalLine();
+        updateFile();
+    }
+
+    public static void updateFile() {
+        try {
+            writeTasksToFile();
+        } catch (IOException e) {
+            // relative path is different for testing
+            System.out.println("Check file path!");
```

Comment:
I'm not sure how the comment relates to the statement.

Reply:
Previously i created the data file differently, so i wrote this comment to remind myself of potential problem during testing. Forgot to remove the comment when it is fixed.